### ISSUE TITLE
All farmland functions as a cellar/enclosed room

### ISSUE THREAD

**Issue by DianaNites** [anegostudios/VintageStory-Issues#4407]
### Game Version

v1.19.8

### Platform

Linux

### Modded

Vanilla

### SP/MP

Singleplayer

### Description

I use a status HUD mod which among other things shows when i'm in a room, and was walking over my farm and noticed the cellar icon. This prompted me to remove all mods and do further testing, checking with `/debug rooms hi` and `/debug rooms list`, and testing some newly created worlds without any mods ever installed.

It appears all farmland is detected as, and functions as, an enclosed room and cellar. I assume its cellar aspects are unusable because items cannot be placed inside farmland, only the player/entities? are capable of being inside it, i assume due to its reduced block height.

I verified this with both the debug commands, and creating a world with an icy start, removing my characters clothes, and and seeing my temperature start to drop outside in the snow, and start to go back up when standing "inside a room" on top of the farmland.

note: i initially thought it was only farmland directly north of water, then testing showed this happen with any direction around water, and then more testing showed it with any farmland at all, and that it was a genuine effect and not the debug command being incorrect.

### How to reproduce

1. Create a new world with creative enabled(or survival and make one)
2. Get a hoe and soil
3. Find or Place soil
4. Till the soil
5. Use `/debug rooms` or functional tests to verify standing on farmland counts as being in a room, specifically a cellar.

### Screenshots

<details>

<summary>Drop from 36.6 to 36.5</summary>

![2024-10-31_00-16-17](https://github.com/user-attachments/assets/7a16c40b-8ca0-4fc1-9c80-653ef4758156)
![2024-10-31_00-16-23](https://github.com/user-attachments/assets/ab1f7b4a-3d53-4bd3-bb84-12260285d176)

</details>

<details>

<summary>Growth from 36.5 all the way back to 37</summary>

![2024-10-31_00-18-45](https://github.com/user-attachments/assets/ce21f390-5a36-42d4-839f-59f6b5208533)
![2024-10-31_00-19-11](https://github.com/user-attachments/assets/dcc1675a-f848-45ab-99fa-e8b7f433299f)
![2024-10-31_00-19-15](https://github.com/user-attachments/assets/bf933eee-a766-48dc-a289-57a322d9801d)

</details>

**Comment by pizza2004** [comment]
Yeah, this is a thing that happens in the engine.  I believe you could experience the same phenomenon with 1.19 trapdoors and with lead pane windows, but I'm not 100% sure.  It's not entirely clear what the proper "fix" would be to this, but it's definitely a bit odd to just stand on farmland and warm up.

**Comment by darkmage0707077** [comment]
Confirmed it's fixed now in 1.21.1. Thank you so much!
